52_wa_diff_shipment.py

```diff
@@ -0,0 +1,263 @@
+# importing the required libraries
+import pandas as pd
+import numpy as np
+import altair as alt
+import statsmodels.api as sm
+import statsmodels.formula.api as smf
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+# loading data for drug prescriptions in Florida and neighbouring states
+prescriptions = pd.read_csv("05_cleaned_data/prescriptions_wa.csv", low_memory=False)
+
+# reducing the number of columns in the drug prescriptions dataset by including only the relevant attributes to create a new dataset
+
+prescriptions_reduced = prescriptions[
+    [
+        "DRUG_CODE",
+        "DRUG_NAME",
+        "QUANTITY",
+        "UNIT",
+        "STRENGTH",
+        "CALC_BASE_WT_IN_GM",
+        "DOSAGE_UNIT",
+        "Product_Name",
+        "Ingredient_Name",
+        "Measure",
+        "MME_Conversion_Factor",
+        "dos_str",
+        "Year",
+        "Month",
+        "StateFIPS",
+        "StateName",
+        "CountyFIPS",
+        "state_abbrev",
+        "FIP_unique",
+        "Population",
+        "county_test",
+    ]
+]
+
+# creating a copy of reduced dataset of prescriptions and converting some of the attributes to appropriate data type
+
+prescriptions_reduced_copy = prescriptions_reduced.copy()
+
+prescriptions_reduced_copy["Year"] = prescriptions_reduced_copy["Year"].astype("int64")
+prescriptions_reduced_copy["DRUG_CODE"] = prescriptions_reduced_copy[
+    "DRUG_CODE"
+].astype("int64")
+prescriptions_reduced_copy["Month"] = prescriptions_reduced_copy["Month"].astype(
+    "int64"
+)
+prescriptions_reduced_copy["Population"] = prescriptions_reduced_copy[
+    "Population"
+].astype("int64")
+
+# creating a dataset that has all the drug prescriptions in the state of Florida
+
+washington_prescriptions = prescriptions_reduced_copy[
+    prescriptions_reduced_copy["StateName"] == "Washington"
+]
+# washington_prescriptions_copy = washington_prescriptions.copy()
+washington_prescriptions["shipment_per_100k"] = (
+    (
+        washington_prescriptions["dos_str"]
+        * washington_prescriptions["DOSAGE_UNIT"]
+        * washington_prescriptions["MME_Conversion_Factor"]
+    )
+    / (washington_prescriptions["Population"])
+    * 100000
+)
```

Comment:
You will need to revise the formula. The reason is that using this formula for calculating the volume of opioid shipped, we have been getting around 80K Null Values, which was affecting the analysis.

The new formula that you would want to use -

washington_prescriptions_copy["shipment_per_100k"] = (
(
washington_prescriptions_copy["CALC_BASE_WT_IN_GM"]*
washington_prescriptions_copy["MME_Conversion_Factor"]
)
/ (washington_prescriptions_copy["Population"])
*100000
)

This greatly reduces the number of null values present for total volume of opioid shipped.

Reply:
Revised in the new ipynb notebook